app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt

```diff
@@ -306,6 +326,63 @@ class StoryFragmentTest {
     }
   }
 
+  @Test
+  fun testStoryFragment_chapterMissingPrerequisiteThumbnailIsBlurred() {
+    launch<StoryActivity>(createFractionsStoryActivityIntent()).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(allOf(withId(R.id.story_chapter_list))).perform(
+        scrollToPosition<RecyclerView.ViewHolder>(
+          2
+        )
+      )
+
+      testCoroutineDispatchers.advanceUntilIdle()
+      onView(
+        atPositionOnView(
+          R.id.story_chapter_list,
+          2,
+          R.id.chapter_thumbnail
+        )
+      ).check { view, noViewFoundException ->
+        Mockito.verify(view.chapter_thumbnail.imageLoader, times(2)).loadDrawable(
+          anyInt(),
+          anyOrNull(),
+          capture(listCaptor)
+        )
+        assertThat(ImageTransformation.BLUR).isIn(listCaptor.value)
+      }
+    }
+  }
+
+  @Test
+  fun testStoryFragment_changeConfiguration_chapterMissingPrerequisiteThumbnailIsBlurred() {
+    launch<StoryActivity>(createFractionsStoryActivityIntent()).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      onView(allOf(withId(R.id.story_chapter_list))).perform(
+        scrollToPosition<RecyclerView.ViewHolder>(
+          2
+        )
+      )
+
+      testCoroutineDispatchers.advanceUntilIdle()
```

Comment:
Is advanceUntilIdle still needed or is runCurrent sufficient? (Prefer the latter if so, and if the former please add a brief one-line comment explaining what timing thing we're waiting for here).

Ditto above.

Reply:
Test passes without advanceUntilIdle, so I think it's ok to remove it.

Reply:
Sounds good--thanks for checking.